### PR TITLE
Fix #10155 include writer outputstream

### LINKED ISSUE #10155: EE10 Servlet include after `HttpServletResponse.getWriter().println()` omits `Content-Length` from the response
**Jetty version(s)**
12

**Description**
Any EE10 servlet that does the following:
```java
response.getWriter().println("Include:");
getServletContext().getRequestDispatcher("/test.txt").include(request, response);
```
will not add the `Content-Length` header to the response.

### PR DESCRIPTION
**Pull request by gregw**
Handle including when there is a mix of input and output streams

### CODE REVIEW

**Comment by gregw** [comment]
This also includes the commits from #10154

**Comment by joakime** [comment]
This PR has 2 testcase failures.

* org.eclipse.jetty.ee10.jstl.JspIncludeTest.testTopWithIncluded
* org.eclipse.jetty.ee10.servlet.IncludedServletTest.testTopWithIncludedHeader

File: jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/IncludedServletTest.java [hunk 1]
```diff
@@ -172,6 +172,9 @@ public void testTopWithIncludedHeader() throws IOException
             IO.copy(reader, writer);
 
             String response = writer.toString();
+            connection.getHeaderFields().entrySet().forEach(System.err::println);
```
Comment (lorban) [inline jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/IncludedServletTest.java:175]: These `System.err.println` should go away.

File: jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DispatcherTest.java [hunk 2]
```diff
@@ -1078,22 +1239,50 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
+            if (useWriter)
+                response.getWriter().println("---");
+            else
+                response.getOutputStream().write("---\n".getBytes(StandardCharsets.US_ASCII));
+        }
+    }
+
+    public static class HelloServlet extends HttpServlet implements Servlet
+    {
+        // The logic linked to this field be deleted and the writer always used once #10155 is fixed.
```
Comment (lorban) [inline jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DispatcherTest.java]: If you want to keep the `useWriter` logic, this comment should go away.

File: jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DispatcherTest.java [hunk 3]
```diff
@@ -380,10 +384,165 @@ public void testInclude() throws Exception
...
+
+        String responses = _connector.getResponse("""
+            GET /context/IncludeServlet?do=static HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+            \r
+            """);
+
+        String expected = """
+            HTTP/1.1 200 OK\r
```
Comment (lorban) [inline jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DispatcherTest.java:508]: What's the reason why using the writer results in the absence of the `Content-Length` header?

**Review by lorban — CHANGES_REQUESTED** [review]

File: jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DispatcherTest.java (hunk 3 above)
Comment (gregw) [reply jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DispatcherTest.java:508]: The setting of a content-length header is highly dependent on the HTTP version, the request version, the request headers and most importantly when/how the response is committed.    We don't want to test if HttpConnection and HttpGenerator are doing the right framing here, that is tested elsewhere.  We just want to see content and/or headers

**Review by lorban — APPROVED** [review]

**Pull request merged**